libs/agno/agno/memory/manager.py

```diff
@@ -707,6 +712,204 @@ def _get_first_n_memories(self, user_id: str, limit: Optional[int] = None) -> Li
 
         return sorted_memories_list
 
+    def optimize_memories(
+        self,
+        user_id: Optional[str] = None,
+        token_limit: int = 2000,
+        strategy: Union[str, MemoryOptimizationStrategyType, MemoryOptimizationStrategy] = "summarize",
+        model: Optional[Model] = None,
+        apply: bool = False,
+    ) -> List[UserMemory]:
+        """Optimize user memories to fit within token limit.
+
+        Args:
+            user_id: User ID to optimize memories for. Defaults to "default".
+            token_limit: Maximum tokens for all memories combined.
```

Comment:
This doesn't work for summarize, because there the token limit is per memory? 

In general I don't think we should have token limit, it doesn't really make sense. The Model should just summarize.